### ISSUE TITLE
NestJS sub-domain and base domain missing

### ISSUE THREAD

**Issue by danialdezfouli** [nestjs/nest#4670]
Subdomains, and base routes are interfering.
other controllers methods ran whole even subdomains

something.localhost:3000 -> Hello World!
localhost:3000 -> Hello World!

app.controller.ts:
```
@Controller()
export class AppController {
  @Get()
  getHello(): string {
    return 'Hello World!';
  }
}
```
another.controller.ts:
```
@Controller({
  host: ':account', // ':account.localhost' -> also tested 
})
export class AnotherController {
  @Get()
  index(@HostParam('account') account: string) {
    return 'running from sub domain';
  }
}
```

## Environment

<pre><code>
Nest version: 7.0.0

 
For Tooling issues:
- Node version: 12.16.1
- Platform:  Linux
</code></pre>

**Comment by danialdezfouli** [comment]
Shall I add default host value to all controllers ?
```
@Controller({
  host: 'localhost',
})
```

**Comment by kamilmysliwiec** [comment]
Express and Fastify don't have built-in filtering by hostname. What this:

```
@Controller({
  host: ':account', // ':account.localhost' -> also tested 
})
```
essentially does is wrapping all your route handlers within a condition which compares the hostname. However, if your routes **without** hostname specified overlap with those which you have under your "hostname"-specific controller + they were registered first, there's no reason to skip them. They will be simply evaluated first and will never hit the second controller.

TL;DR - routes with hostname filtering should not overlap with routes without filtering.

**Comment by danialdezfouli** [comment]
Thanks for your reply, so you confirm my second comment.
1. it might bee good to add a default host to all controllers.
2. How to handle localhost keyword in dev and real domain in prod mode ?

I should use allways the same function as shown bellow ?
const host = () => process.env.NODE_ENV === 'development' ? 'localhost' : 'example.com'

**Comment by kamilmysliwiec** [comment]
> it might bee good to add a default host to all controllers.

Correct (unless they don't overlap with routes that have filtering by host applied).

> How to handle localhost keyword in dev and real domain in prod mode ?

You can use env vars as you proposed above
